internal/scm/github/github.go

```diff
@@ -189,7 +193,10 @@ func (g *Github) GetRepositories(ctx context.Context) ([]scm.Repository, error)
 
 func (g *Github) getRepositories(ctx context.Context) ([]*github.Repository, error) {
 	allRepos := []*github.Repository{}
-
+	var repoRegEx *regexp.Regexp // repoFilter flag will filter all collected repositories
+	if g.RepositoryFilter != "" {
+		repoRegEx = regexp.MustCompile(g.RepositoryFilter)
+	}
```

Comment:
Consider encapsulating the compilation and matching in a method, using a field to hold state, like 

```
func (g *Github) matchesRepositoryFilter(n string)
  if g.RepositoryFilter == "" {
    return true
  }
  if g.compiledRepositoryFilter == nil {
    g.compiledRepositoryFilter = regexp.MustCompile(g.RepositoryFilter)
  }

  return repoRegEx.MatchString(n)
}
```

This returns true if there is no filter (all included) otherwise caches and uses the compiled regex, making the below logic less busy.

Reply:
Thanks for the feedback! Will refactor with the above suggestions, I think a `repo-include` and `repo-exclude` is a great implementation. Especially since Go currently does not support `negative lookahead` in regular expressions.

Reply:
created a similar listener since `repo-exclude` is used to remove matching patterns whereas `repo-include` accepts matching patterns. I think this helps de-couple logic but open to suggestions.